cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -0,0 +1,295 @@
+
+    #endif
+
+    return fixed_point<Rep1, Rad1>{scaled_integer<Rep1>(lhsv + rhsv, scale)};
+}
+
+// MINUS Operation
+template<typename Rep1, Radix Rad1,
+         typename Rep2, Radix Rad2>
+fixed_point<Rep1, Rad1> operator-(fixed_point<Rep1, Rad1> const& lhs,
+                                  fixed_point<Rep2, Rad2> const& rhs) {
+
+    static_assert(std::is_same<Rep1, Rep2>::value, "Represenation types should be the same");
+    static_assert(Rad1 == Rad2,                    "Radix types should be the same");
+
+    auto const rhsv  = lhs._scale > rhs._scale ? detail::shift<Rad1>(rhs._value, scale_type{lhs._scale - rhs._scale}) : rhs._value;
+    auto const lhsv  = lhs._scale < rhs._scale ? detail::shift<Rad1>(lhs._value, scale_type{rhs._scale - lhs._scale}) : lhs._value;
+    auto const scale = lhs._scale > rhs._scale ? lhs._scale : rhs._scale;
+
+    #if defined(__CUDACC_DEBUG__)
+
+    assert(!subtraction_overflow<Rep1>(lhsv, rhsv) &&
+        "fixed_point overflow of underlying represenation type " + print_rep<Rep1>());
+
+    #endif
+
+    return fixed_point<Rep1, Rad1>{scaled_integer<Rep1>(lhsv - rhsv, scale)};
+}
+
+// MULTIPLIES Operation
+template<typename Rep1, Radix Rad1,
+         typename Rep2, Radix Rad2>
+fixed_point<Rep1, Rad1> operator*(fixed_point<Rep1, Rad1> const& lhs,
+                                  fixed_point<Rep2, Rad2> const& rhs) {
+
+    static_assert(std::is_same<Rep1, Rep2>::value, "Represenation types should be the same");
+    static_assert(Rad1 == Rad2,                    "Radix types should be the same");
+
+    #if defined(__CUDACC_DEBUG__)
+
+    assert(!multiplication_overflow<Rep1>(lhs._value, rhs._value) &&
+        "fixed_point overflow of underlying represenation type " + print_rep<Rep1>());
+
+    #endif
+
+    return fixed_point<Rep1, Rad1>{scaled_integer<Rep1>(lhs._value * rhs._value, scale_type{lhs._scale + rhs._scale})};
+}
+
+// DIVISION Operation
+template<typename Rep1, Radix Rad1,
+         typename Rep2, Radix Rad2>
+fixed_point<Rep1, Rad1> operator/(fixed_point<Rep1, Rad1> const& lhs,
+                                  fixed_point<Rep2, Rad2> const& rhs) {
+
+    static_assert(std::is_same<Rep1, Rep2>::value, "Represenation types should be the same");
+    static_assert(Rad1 == Rad2,                    "Radix types should be the same");
+
+    #if defined(__CUDACC_DEBUG__)
+
+    assert(!division_overflow<Rep1>(lhs._value, rhs._value) &&
+        "fixed_point overflow of underlying represenation type " + print_rep<Rep1>());
+
+    #endif
+
+    return fixed_point<Rep1, Rad1>{scaled_integer<Rep1>(lhs._value / rhs._value, scale_type{lhs._scale - rhs._scale})};
+}
+
+// EQUALITY COMPARISON Operation
+template<typename Rep1, Radix Rad1,
+         typename Rep2, Radix Rad2>
+bool operator==(fixed_point<Rep1, Rad1> const& lhs,
+                fixed_point<Rep2, Rad2> const& rhs) {
+    return lhs.get() == rhs.get();
```

Comment:
Since `get()` returns double, maybe it would be a good idea to use an epsilon rather than direct equality, like in PR https://github.com/rapidsai/cudf/pull/3776

Reply:
I have added the epsilon change similar to 3776. As for your comment Chris, not sure I completely understand but remember that when when your scales are different, you need to either shift them to the same scale first, to use the `get` method (which is shifting as well), which is what I opted for.

Reply:
In the same way that `operator==` for floats doesn't use an epsilon, we shouldn't be using an epsilon in an `operator==`. That will cause surprising behavior. 

If comparison with an epsilon is desired, that should be done in a free function where the epsilon can be specified, not in an `operator==`.